src/overlays/actors/ovl_En_Hit_Tag/z_en_hit_tag.c

```diff
@@ -29,21 +28,66 @@ const ActorInit En_Hit_Tag_InitVars = {
     (ActorFunc)NULL,
 };
 
-// static ColliderCylinderInit sCylinderInit = {
-static ColliderCylinderInit D_80BE21F0 = {
-    { COLTYPE_NONE, AT_NONE, AC_ON | AC_TYPE_PLAYER, OC1_ON | OC1_TYPE_PLAYER, OC2_TYPE_1, COLSHAPE_CYLINDER, },
-    { ELEMTYPE_UNK0, { 0x00000000, 0x00, 0x00 }, { 0xF7CFFFFF, 0x00, 0x00 }, TOUCH_NONE | TOUCH_SFX_NORMAL, BUMP_ON, OCELEM_NONE, },
+static ColliderCylinderInit sCylinderInit = {
+    {
+        COLTYPE_NONE,
+        AT_NONE,
+        AC_ON | AC_TYPE_PLAYER,
+        OC1_ON | OC1_TYPE_PLAYER,
+        OC2_TYPE_1,
+        COLSHAPE_CYLINDER,
+    },
+    {
+        ELEMTYPE_UNK0,
+        { 0x00000000, 0x00, 0x00 },
+        { 0xF7CFFFFF, 0x00, 0x00 },
+        TOUCH_NONE | TOUCH_SFX_NORMAL,
+        BUMP_ON,
+        OCELEM_NONE,
+    },
     { 16, 32, 0, { 0, 0, 0 } },
 };
 
-#endif
+void EnHitTag_Init(Actor* thisx, GlobalContext* globalCtx) {
+    ColliderCylinder* pCylinder;
+    EnHitTag* this = (EnHitTag*)thisx;
 
-extern ColliderCylinderInit D_80BE21F0;
+    Actor_SetScale(&this->actor, 1.0f);
+    this->actionFunc = func_80BE20E8;
+    pCylinder = &this->cylinder;
+    Collider_InitAndSetCylinder(globalCtx, &this->cylinder, &this->actor, &sCylinderInit);
+    Collider_UpdateCylinder(&this->actor, &this->cylinder);
+    if (Flags_GetSwitch(globalCtx, (s32)(this->actor.params & 0xFE00) >> 9)) {
```

Comment:
It's not matching when I remove the s32 cast. I'm aware about using macros to access actor params, but I don't know what this param is so I'm not sure what to call the macro. Is it a good idea to create a macro even if it just uses a placeholder name? Thanks for the comment.

Reply:
Its a switch flag, each scene has a large bitfield of them to detect if switches have previously been flipped. The actor is passed this value to use to check if the scene switch has already been flipped, ergo, if the hit mark has already been triggered by the player. 

This is to stop the player from hitting a hit spot, leaving the area, and coming back to hit it again, it add persistence by making the actor check when it spawns in a scene "has this spot already been hit before? if yes, just die so we can't be hit again (until Song of time)"

And yes, it's currently requested that all math/shifted params have macros even if you don't know what they are doing. Off my head it would be named something like "ENHITTAG_GET_FE00" or similar
example: `#define ENST_GET_3F(thisx) (((thisx)->params & 0x3F) & 0xFF)`
although in this case we know what it does so
`#define ENST_GET_SWITCHFLAG(thisx) (((s32)(thisx)->params & 0xFE00) > 0x9)` < didnt check if this matches

Reply:
Thanks very much for the explanation, have added this